### ISSUE TITLE
What's the difference between encode() and tokenizer()+model()?

### ISSUE THREAD

**Issue by HugoHann** [UKPLab/sentence-transformers#1317]
When I run:
```
from sentence_transformers import SentenceTransformer
sentences = [ "Each sentence is converted"]
model = SentenceTransformer('paraphrase-MiniLM-L6-v2')
embeddings = model.encode(sentences)
```

and after I run:
```
tokenizer = AutoTokenizer.from_pretrained('paraphrase-MiniLM-L6-v2')
model = AutoModel.from_pretrained('paraphrase-MiniLM-L6-v2')
inputs = tokenizer(sentences, padding=True, truncation=True, return_tensors="pt")
with torch.no_grad():
    embd=model(**inputs,output_hidden_states=True, return_dict=True).pooler_output
```

It has totally different output, why? I've checked the encode(), didn't find the reason. I even add normlization after the latter, but it has  another output either.
 another example, I output their sum:
```
from transformers import AutoTokenizer, AutoModel
import torch
from sentence_transformers import SentenceTransformer
import numpy as np
if __name__ == '__main__':
    def mean_pooling(model_output, attention_mask):
        token_embeddings = model_output[0] #First element of model_output contains all token embeddings
        input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
        return torch.sum(token_embeddings * input_mask_expanded, 1) / torch.clamp(input_mask_expanded.sum(1), min=1e-9)
    sentences = ['This is an example sentence', 'Each sentence is converted']

    tokenizer = AutoTokenizer.from_pretrained('./pretrained/all-mpnet-base-v1')
    model = AutoModel.from_pretrained('./pretrained/all-mpnet-base-v1')

    encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt')

    with torch.no_grad():
        model_output = model(**encoded_input)

    sentence_embeddings = mean_pooling(model_output, encoded_input['attention_mask'])

    print("Hugging Face:")
    print(torch.sum(sentence_embeddings[0]))


    model = SentenceTransformer('./pretrained/all-mpnet-base-v1')
    embeddings = model.encode(sentences)
    print("Sentence Transformers:")
    print(np.sum(embeddings[0]))

```
output:
```
Hugging Face:
tensor(0.2206)
Sentence Transformers:
0.09434402
```

Could somebody help me please?

**Comment by nreimers** [comment]
You can find an example here:
https://huggingface.co/sentence-transformers/paraphrase-MiniLM-L6-v2#usage-huggingface-transformers

The model uses mean_pooling, not the returned .pooler_output value as you have it in your first example.


For the all-* model: 
Embeddings are normalized by sentence-transformers, what you need to do by hand when you just use transformers. See:
https://huggingface.co/sentence-transformers/all-mpnet-base-v2#usage-huggingface-transformers

**Comment by HugoHann** [comment]
> You can find an example here: https://huggingface.co/sentence-transformers/paraphrase-MiniLM-L6-v2#usage-huggingface-transformers
> 
> The model uses mean_pooling, not the returned .pooler_output value as you have it in your first example.
> 
> For the all-* model: Embeddings are normalized by sentence-transformers, what you need to do by hand when you just use transformers. See: https://huggingface.co/sentence-transformers/all-mpnet-base-v2#usage-huggingface-transformers

Thank you for replying! 
Yep, after add the normalize after mean_pooling, they output the same results except the precision.

**Comment by MrRace** [comment]
@nreimers If I would like to get to pooler_output from `model = SentenceTransformer('./pretrained/all-mpnet-base-v1')` like `embd=model(**inputs,output_hidden_states=True, return_dict=True).pooler_output`. Is that possible？ And How should I change？

**Comment by nreimers** [comment]
@MrRace You can load them with normal huggingface transfromers code
